modules/exploits/windows/http/apache_tika_jp2_jscript.rb

```diff
@@ -0,0 +1,99 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Powershell
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache Tika Header Command Injection',
+      'Description'    => %q{
+          This module exploits a command injection vulnerability in Apache
+        Tika 1.7-1.17 on Windows.  A file with the image/jp2 content-type is
+        used to bypass magic bytes checking.  When OCR is specified in the
+        request, parameters can be passed to change the parameters passed
+        at command line to allow for arbitrary JScript to execute.  A
+        JScript stub is passed to execute arbitrary code.
+        When a This module was verified against version
+        1.17 on Windows 2012.
+      },
+      'License'        => MSF_LICENSE,
+      'Privileged'     => false,
+      'Platform'       => 'win',
```

Comment:
I know this was written for Windows, but is there any reason a Unix system couldn't be targeted?

Reply:
Not sure. it calls a .exe on windows and passes params to it.  If the .exe equivalent on linux has the same parameters, there's a good chance the flags would match.  All the documentation was for windows.
Devils in the details.  I thought about getting this written first, then if time permitted (most likely wouldn't) going back to see if it works for additional OSes

Reply:
Easily tested. I wanted someone to look into that. cc @sgonzalez-r7

Reply:
> I know this was written for Windows, but is there any reason a Unix system couldn't be targeted?

I did a little testing on Unix when working on this and was only able to break out of the command on Windows. The command in the end looks like `"injected.exe"originalcommand.exe some args here`. The way Windows intereprets double quote is what lets the injection happen.